packet_link_qualification/packet_link_qualification.proto

```diff
@@ -70,98 +78,92 @@ enum QualificationState {
-    // of the test and ensure all packets circulating as part of the test have
-    // been dropped.
-    int32 minimum_wait_post_qualification_seconds = 9;
-    
-    // default packet_size is 1500 bytes.
-    uint16 packet_size = 10;
-    
-  }
+enum QualificationEnd {
+  END_UNSPECIFIED = 0;
+  // Responsible for generating loopback packets, etc.
+  END_NEAR = 1;
+  // Responsible for looping back packets only.
+  END_FAR = 2;
+}
 
+message StartPacketQualificationRequest {
   // Per interface start configuration request.
-  repeated QualificationConfiguration configs = 1;
+  repeated QualificationConfiguration config = 1;
+}
+
+message QualificationConfiguration {
+  // The ID assigned to the given test. The ID value must be unique among the
+  // tests running on this agent.
+  string id = 1;
+
+  // Path to the interface corresponding to the port to test.
+  types.Path interface = 2;
+
+  // The minimum amount of wait time before sending packets on the link. It
+  // gives peer additional time to prepare. The default value for this field
+  // is 70 seconds.
+  int32 minimum_wait_before_preparation_seconds = 3;
+
+  // The total number of packets to send on the link in the preparation phase.
+  int64 num_preparation_packets = 4;
+
+  // The amount of time in which the preparation must complete.
+  int32 preparation_timeout_seconds = 5;
+
+  // The amount of time the qualification will run before we recapture the
+  // packets and check if there was any drop.
+  int32 qualification_duration_seconds = 6;
+
+  // Qualification end.
+  QualificationEnd qualification_end = 7;
+
+  // The default value for this field is 10 seconds. This is the
+  // time period that the link qualification service should wait
+  // between the two samples of port counters that it reads from the
+  // device - both prior to starting qualification, and right after
+  // qualification. The qualification itself runs for
+  // Qaulification_duration_seconds.
+  int32 test_wait_between_sample_seconds = 8;
+
+  // The default value of this field is 10 seconds. During this period, the
+  // link qualification state machine must undo things it setup during start
+  // of the test and ensure all packets circulating as part of the test have
+  // been dropped.
+  int32 minimum_wait_post_qualification_seconds = 9;
+
+  // default packet_size is 1500 bytes.
+  uint32 packet_size = 10;
+
 }
 
 message StartPacketQualificationResponse {
   // Per interface start response.
-  message StartResponse {
-    // The same ID given in the start request.
-    string id = 1;
-    // Status of the operation. In case of failures, canonical error code and
-    // message will show the details.
-    google.rpc.Status status = 2;
-  }
-
-  repeated StartResponse responses = 1;
+  repeated StartResponse response = 1;
+}
+
+message StartResponse {
+  // The same ID given in the start request.
+  string id = 1;
 }
 
 message StopPacketQualificationRequest {
```

Comment:
should this be cancel? vs. stop?

Reply:
what it is noted above that stop clears the stats - that feels conterinitive from a workflow

seems like it would be

Setup
Start
Get (to see if things are progressing) (may not be available)
Stop (may not be available)
Get (once the system reachability has been guaranteed to be restored
Clear